lib/http/node.js

```diff
@@ -21,112 +22,96 @@ const Logger = require('../logger');
  * @return  {Number} number of milliseconds to wait before retrying again the request.
  */
 NodeHttpClient.prototype.constructExponentialBackoffStrategy = function () {
-  var sleep = this._connectionConfig.getRetrySfStartingSleepTime();
-  var cap = this._connectionConfig.getRetrySfMaxSleepTime();
-  var base = 1;
+  let sleep = this._connectionConfig.getRetrySfStartingSleepTime();
+  const cap = this._connectionConfig.getRetrySfMaxSleepTime();
+  const base = 1;
   sleep = Util.nextSleepTime(base, cap, sleep);
   return sleep * 1000;
-}
+};
 
 /**
  * Creates a client that can be used to make requests in Node.js.
  *
  * @param {ConnectionConfig} connectionConfig
  * @constructor
  */
-function NodeHttpClient(connectionConfig)
-{
+function NodeHttpClient(connectionConfig) {
   Base.apply(this, arguments);
 }
 
 Util.inherits(NodeHttpClient, Base);
 
-/**
- * @inheritDoc
- */
-NodeHttpClient.prototype.getAgentAndProxyOptions = function (url, proxy, mock)
-{
-  const isHttps = Url.parse(url).protocol === 'https:';
-  let agentClass;
-  let agentOptions = { keepAlive: true };
-  let options;
-  var bypassProxy = false;
+const httpsAgentCache = new Map();
+
+function getFromCacheOrCreate(agentClass, options, parsedUrl) {
+  const protocol = parsedUrl.protocol || 'http:';
```

Comment:
are you sure that `http:` should be the default protocol?

Reply:
Probably we don't need default here.